Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM rust:1.60.0 as rustbuilder
+WORKDIR /tools
+RUN git clone https://github.com/parallaxsecond/parsec-tool.git
```

Comment:
Agree, but not possible yet because the demo makes use of the `encrypt` command, which was only just added and is not yet in a release.

Reply:
You can use a specific commit then

Reply:
I generally dislike commit hashes, but yes I can do that.

Reply:
It looks like Git doesn't allow you to specify a commit ID as part of cloning? (Or am I missing something?) This means I'd actually need to `checkout` the commit, which means modifying the command sequence post-cloning, which I'd rather not do. Once we release `parsec-tool` we can use a specific version with the level of support that we need. I'd rather wait until then. I can cover with a GH issue if it's that important.

Reply:
It's a standard practice to do and use everywhere:
git clone <repo>
git checkout <tag/commit/branch>
You will have to do it anyway to obtain the correct release version.

Reply:
But when we have a release version, it can be done as part of the `clone`, which is my point. The intent of that would be very clear. The intent of having an extra command to checkout a specific untagged hash is not clear at all.

Reply:
No, it won't be a part of the clone. git clone always gets the latest commit. You will have to checkout the released version anyway

Reply:
You can clone a tag/branch like [this](https://github.com/parallaxsecond/parsec/blob/main/e2e_tests/docker_image/parsec-service-test-all.Dockerfile#L22)

Reply:
Oh.. Perhaps I misunderstand, but my plan would have been to use `clone` with `--branch` similar to how we check out the correct version of TSS here: https://github.com/parallaxsecond/parsec/blob/006c8d66ddeab678e978cc50cbcd894d1226590e/e2e_tests/docker_image/parsec-service-test-all.Dockerfile#L22 

But that only works once you have a tag. You can't do that with a commit.
Doing `checkout` would work either way, but the intent is much less clear when it's just some random hash.

Reply:
When a commit hash is used usually there is a comment about the commit.
The comment will be removed when we replace the commit hash with a release tag

Reply:
Okay, changed.